### PR TITLE
Finishing up P107 (has a closed point), S46, S47

### PR DESCRIPTION
**Pull request by yhx-12243**

### CODE REVIEW

**Comment by prabau** [comment]
Sorry about the following, but do you have write access to the official repository?  If so, do you mind closing this PR and redoing it on a branch in the official repo?  That will make it available in preview mode from the web site, also easier to review.

**Comment by yhx-12243** [comment]
> Sorry about the following, but do you have write access to the official repository? If so, do you mind closing this PR and redoing it on a branch in the official repo? That will make it available in preview mode from the web site, also easier to review.

Done.

**Comment by prabau** [comment]
I think you have to close the current PR and open a new one.  One cannot change the branch for an existing PR.

**Comment by prabau** [comment]
(looking at the branch `p107` from the pi-base web site)

P46: there is now circular dependency in the reasoning between path connected and arc connected.
Same thing for the "locally ..." versions.

Also the justification
> For every point $x$, there are $\mathfrak c$ many points $y$ topologically indistinguishable from $x$. So any path in the space can be turned into an arc.

is not completely clear to me.  Why is that exactly?  If the reasoning is not obvious, it seems it would be worth a post on mathse asking/explaining the technique in general.

File: spaces/S000047/properties/P000126.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+space: S000047
+property: P000126
+value: false
+---
+
+$\{2, 3\}$ and its complement are neither open nor closed.
```
Comment (prabau) [inline spaces/S000047/properties/P000126.md:7]:
```suggestion
$\{2, 3\}$ is neither open nor closed.
```

**Comment by prabau** [comment]
S48: I don't think $\\{p\\}$ is closed.  But for every $n\in\omega$ the singleton $\\{n\\}$ is closed.

File: theorems/T000649.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+uid: T000649
+if:
+  and:
+  - P000126: true
+  - P000137: false
+then:
+  P000107: true
+---
+
+If a non-empty {P126} space has no closed points, then every singleton is open, which implies that the space is {P52}, a contradiction.
```
Comment (prabau) [inline theorems/T000649.md:11]:
```suggestion
If a {P126} space has no closed points, then every singleton is open, which implies that the space is {P52}.
This is a contradiction if the space is non-empty.
```

**Comment by prabau** [comment]
T649: That's a nice little result.  Allows to derive
https://topology.pi-base.org/spaces/S000082/properties/P000126.

**Comment by yhx-12243** [comment]
> (looking at the branch `p107` from the pi-base web site)
> 
> P46: there is now circular dependency in the reasoning between path connected and arc connected. Same thing for the "locally ..." versions.
> 
> Also the justification
> 
> > For every point $x$, there are $\mathfrak c$ many points $y$ topologically indistinguishable from $x$. So any path in the space can be turned into an arc.
> 
> is not completely clear to me. Why is that exactly? If the reasoning is not obvious, it seems it would be worth a post on mathse asking/explaining the technique in general.

First, the circular dependencies is the issue of the pi-base deduction system, because we already have path connected derived from [contractible](https://topology.pi-base.org/spaces/S000046/properties/P000199), then my argument is that, for every point $x$, we have continuum many points that is topologically indistinguishable with them, so we can change any non-injective path into the injective one by making each fiber $f^{-1}(x)$ injective.

**Comment by david20000813** [comment]
Perhaps, to avoid the appearance of circular reasoning, we can add the original argument back in? For path-connected, just note that the space is contractible and cite T583 and T590. For locally path-connected, just note the space is Alexandrov and cite T316.

**Comment by prabau** [comment]
Right.  I will suggest something.

**Comment by prabau** [comment]
@yhx-12243 You made a bunch of updates.
I suggested to create a new PR with the new branch `p107`, but for some reason you did not want to do this.
Please update the `p107` branch to make things easier to look at and review.

**Comment by prabau** [comment]
> Perhaps, to avoid the appearance of circular reasoning, we can add the original argument back in? For path-connected, just note that the space is contractible and cite T583 and T590. For locally path-connected, just note the space is Alexandrov and cite T316.

The recommended way to do this is not to mention specific theorems (which may change at any time), but rely on pi-base itself to do the deduction work.  Searching for `(Explore)` in the source gives several examples.  For example:
https://topology.pi-base.org/theorems/T000519.

So for example for S46-P38 (arc connected), we want to say in summary:
"the space is contractible, hence path connected. then blah blah argument to contruct an injective path."

So use something like, to take advantage of pi-base itself:
`{S46|P199}, hence is {P37} [(Explore)](https://topology.pi-base.org/spaces?q=Contractible%2B%7EPath+connected).`
followed by the rest of the argument.

**Comment by prabau** [comment]
Now we really want to see all that on a branch in the official repo.  So I will close this PR and you recreate another one.

**Comment by prabau** [comment]
As for the other part of the argument (from path to arc), see
https://math.stackexchange.com/questions/5006471/situation-where-a-non-t-0-path-connected-space-is-arc-connected

which you or someone else can provide an answer for.

**Comment by prabau** [comment]
@yhx-12243  As for the other part of the argument (from path to arc), see
https://math.stackexchange.com/questions/5006471/situation-where-a-non-t-0-path-connected-space-is-arc-connected

which you or someone else can provide an answer for.

**Pull request closed without merging**